addons/website_blog/controllers/main.py

```diff
@@ -184,11 +184,12 @@ def blog(self, blog=None, tag=None, page=1, search=None, **opt):
 
         blogs = Blog.search(request.website.website_domain(), order="create_date asc, id asc")
 
-        if not blog and len(blogs) == 1:
-            return werkzeug.utils.redirect('/blog/%s' % slug(blogs[0]), code=302)
-
         date_begin, date_end, state = opt.get('date_begin'), opt.get('date_end'), opt.get('state')
 
+        if not blog and len(blogs) == 1:
+            url = QueryURL('/blog/%s' % slug(blogs[0]), date_begin=date_begin, date_end=date_end, search=search, **opt)()
+            return werkzeug.utils.redirect(url, code=302)
```

Comment:
Can't you push this code back to where it was and avoid duplicating the date_begin/date_end values you pass to QueryURL?

Reply:
Changed